src/read.c

```diff
@@ -1048,6 +1048,49 @@ static avifResult avifDecoderItemValidateAV1(const avifDecoderItem * item, avifD
         return AVIF_RESULT_BMFF_PARSE_FAILED;
     }
 
+    if (!memcmp(item->type, "grid", 4)) {
+        // MIAF (ISO 23000-22:2019), Section 7.3.11.4.1:
+        //   All input images of a grid image item shall use the same coding format, chroma sampling format, and the
+        //   same decoder configuration (see 7.3.6.2).
+        for (uint32_t i = 0; i < item->meta->items.count; ++i) {
+            const avifDecoderItem * tile = &item->meta->items.item[i];
+            if (tile->dimgForID == item->id) {
+                // The coding format is defined by the item type.
+                // avifDecoderGenerateImageGridTiles() skipped any non-AV1 tile.
+                if (memcmp(tile->type, "av01", 4)) {
```

Comment:
Would this skip a heic tile and not fail like the comment seems to imply? *All input images of a grid image item shall use the same coding format...*

Though it's shall and not SHALL...

Reply:
Done in https://github.com/AOMediaCodec/libavif/pull/1393.